helloworld/elementary_4_5.c

```diff
@@ -0,0 +1,32 @@
+#include<stdio.h>
+
+int main() {
+
+    int szam;
+	int osszeg = 0;
+  
+  	setbuf(stdout, NULL);
+	
+	printf("Írj egy számot: ");    
+	scanf("%d", &szam);
+
+	// és akkor most legyen itt ciklussal, mert erre nincs matematikai képlet, bár gondolkozom rajta :-)
+
+	int i;
+	for (i=1; i <= szam; i++)
+	{
+		if (i % 3 == 0 || i % 5 == 0) {
+			
+			osszeg = osszeg + i;
+			
+		}
+	}
```

Comment:
Ha a `<=` operátor köré raksz space-t, akkor a `=` operátor köré is rakj:
`for (i = 1; i <= szam; i++)`

Ha az `int main() {` és az `if (i % 3 == 0 || i % 5 == 0) {`-nél ugyanarra a sorra raktad a nyitó kapcsos zárójelet, akkor for-nál ne kezdd új sorba.

`osszeg = osszeg + i;` helyett írhattad volna azt, hogy `osszeg += i;`

A megoldás jó, és szerintem ezt a választ várta aki kitalálta, de ha már az előzőt meg tudtad oldani jobban, akkor megsúgom, hogy erre is van egy jobb megoldás. A for loop minden egyes számon végigmegy, és minden alkalommal meg kell állapítani if-fel, hogy osztható-e az adott szám 3-al, vagy 5-tel, nem pazarlás ez?

Nem lehetne gyorsabban valahogy két for loop-pal csak a 3-al és az 5-tel osztható számokon végigmenni? és akkor még az if se kéne. Van vele valami bökkenő?

Reply:
Szóval, lehetne azt csinálni, hogy végig megyek a 3-mmal osztható számokon úgy, hogy 3-mmal inkrementálok, aztán az öttel oszthatókon, úgy hogy 5-tel inkrementálok, aztán a 15-tel osztható számokat meg kivonom egy 3. looppal, amit 15-ösével inkrementálok :-). Gyorsabb lesz, de megírni lassabb. Szeretnéd, hogy megcsináljam?